### DISCUSSION TITLE
Setting player's gamemode does not work

### DISCUSSION THREAD

**Discussion by GoldenGamerLP** [Minestom/Minestom#1102] [Help] [answered]
Hello there, the title says it all, im using the latest minestom version from jitpack. I also verified that the code is beeing runned. 
![image](https://user-images.githubusercontent.com/65727384/169647495-d5344ac6-1200-4340-8f44-cfcc7501cb06.png)

After the code has been run: (You see the debug message)
![image](https://user-images.githubusercontent.com/65727384/169647541-0acb56ab-dd04-436a-b4e4-0cac5b0df75c.png)

**Comment by DasLixou** [comment]
Could be that the changes doesn’t get notified. Try to add `player.getEntityMeta().setNotifyAboutChanges(true);` after you set all your changes

**Reply by GoldenGamerLP** [reply]
Doesn't work, the game mode doesn't change client-side but changes on the server side

**Reply by DasLixou** [reply]
When do you execute it? When you use `PlayerLoginEvent`, change it to `PlayerSpawnEvent`

**Reply by GoldenGamerLP** [reply]
I'm not using any event, I'm settings the players game mode in a command

**Comment by VegetalDev** [comment, marked as answer]
It sounds to me like your player's instance isn't set when you're changing the gamemode. If the player doesn't have a set instance, Minestom doesn't appear to broadcast a gamemode update packet. 

You can check this by calling `Player#isActive`, which should return `true` if you're linked to an instance.
